File: App/app/src/main/java/com/example/econo_misons/views/DepenseAdapter.java
package com.example.econo_misons.views;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.econo_misons.database.models.Category;

import java.util.List;

public class DepenseAdapter extends RecyclerView.Adapter<DepenseViewHolder> {

    public List<Category> categories;

    public void updateCategories(List<Category> categoryList){
        categories = categoryList;
        this.notifyDataSetChanged();
    }

    @NonNull
    @Override
    public DepenseViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    public void onBindViewHolder(@NonNull DepenseViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 0;
    }
}

File: MoneySaver/app/src/main/java/fr/hplovers/moneysaver/adapter/DepenseAdapter.kt
package fr.hplovers.moneysaver.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import fr.hplovers.moneysaver.R

class DepenseAdapter(private val layoutId: Int) : RecyclerView.Adapter<DepenseAdapter.ViewHolder>() {

    // On range tout les composants qu'on veut controler
    class ViewHolder(view: View ) : RecyclerView.ViewHolder(view){
        val depenseImage = view.findViewById<ImageView>(R.id.depense_category_image)
    }

    //Injecter notre layout
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater
            .from(parent.context)
            .inflate(layoutId,parent,false)

        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
    }

    //On indique le nombre d'item à afficher dynamiquement (Ici 5)
    override fun getItemCount(): Int = 5
}